Snowflake.Data/Core/ArrowResultSet.cs

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.Arrow.Ipc;
+using Snowflake.Data.Client;
+using Snowflake.Data.Log;
+
+namespace Snowflake.Data.Core
+{
+    class ArrowResultSet : SFBaseResultSet
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<ArrowResultSet>();
+        
+        private int _currentChunkRowIdx = -1;
+        private int _currentChunkRowCount;
+        private readonly int _totalChunkCount;
+        private IResultChunk _currentChunk;
+        private readonly IChunkDownloader _chunkDownloader;
+
+        public ArrowResultSet(QueryExecResponseData responseData, SFStatement sfStatement, CancellationToken cancellationToken) : base()
+        {
+            columnCount = responseData.rowType.Count;
+            try
+            {
+                using (var stream = new MemoryStream(Convert.FromBase64String(responseData.rowsetBase64)))
+                {
+                    using (var reader = new ArrowStreamReader(stream))
+                    {
+                        var recordBatch = reader.ReadNextRecordBatch();
+                        _currentChunkRowCount = recordBatch.Length;
+                        _currentChunk = new ArrowResultChunk(recordBatch);
+                    }
+                }
+
+                this.sfStatement = sfStatement;
+                UpdateSessionStatus(responseData);
+
+                if (responseData.chunks != null)
+                {
+                    _totalChunkCount = responseData.chunks.Count;
+                    
+                    // TODO in SNOW-893835 - support for multiple chunks
```

Comment:
Looks like we are breaking tasks in different JIRA. Make sure ARROW format is not enable until all JIRAs complete.